src/main/java/org/craftercms/studio/impl/v2/utils/DependencyUtils.java

```diff
@@ -43,54 +41,45 @@ public class DependencyUtils {
 	public static final Integer MAX_DEPENDENCY_PATH_LENGTH = 4000;
 
 	/**
-	 * Add the script snippets to update the dependencies for the given path
+	 * Update the dependencies for the given path
 	 *
 	 * @param siteId            the site id
 	 * @param path              the content item path
 	 * @param oldPath           the content item old path
-	 * @param file              the file
 	 * @param dependencyService the dependency service
-	 * @throws IOException if an error occurs while updating the script
+	 * @param sqlSession        the sql session
 	 */
-	public static void addDependenciesScriptSnippets(String siteId, String path, String oldPath,
-							 Path file, DependencyService dependencyService)
-		throws IOException {
-		addDependenciesScriptSnippets(siteId, path, oldPath, file, dependencyService, true, true);
+	public static void updateDependencies(String siteId, String path, String oldPath,
+										  DependencyService dependencyService, SqlSession sqlSession) {
+		updateDependencies(siteId, path, oldPath, dependencyService, sqlSession, true, true);
 	}
 
 	/**
-	 * Add the script snippets to update the dependencies for the given path
+	 * Update the dependencies for the given path
 	 *
 	 * @param siteId            the site id
 	 * @param path              the content item path
 	 * @param oldPath           the content item old path
-	 * @param file              the file
 	 * @param dependencyService the dependency service
+	 * @param sqlSession        the sql session
 	 * @param cleanExisting     if true, the existing dependencies for the path will be deleted
 	 * @param revalidate        if true, the existing dependencies pointing to the path will be set to valid=true
-	 * @throws IOException if an error occurs while updating the script
 	 */
-	public static void addDependenciesScriptSnippets(String siteId, String path, String oldPath,
-							 Path file, DependencyService dependencyService,
-							 boolean cleanExisting, boolean revalidate)
-		throws IOException {
+	public static void updateDependencies(String siteId, String path, String oldPath,
+										  DependencyService dependencyService, SqlSession sqlSession, boolean cleanExisting, boolean revalidate) {
+		DependencyDAO dependencyDao = sqlSession.getMapper(DependencyDAO.class);
```

Comment:
Can we get the mapper in the caller just once? Here it seems we will get multiple `DependencyDAO` mappers, one per updated path.

Reply:
I updated to send `dependencyDao` and `itemDao` from the caller.